File: tasks/2025/round1/wordle2.cpp
#include<iostream>
#include<string>
#include<cassert>

using namespace std;
void tc(){
    string s="abcde",ans="zzzzz";
    string found;
    for(int iter=0;iter<5;iter++){
        cout<<"? "<<s<<endl;
        string resp;
        cin>>resp;
        for(int i=0;i<5;i++){
            if(resp[i]!='W'){
                found.push_back('a'+iter*5+i);
            }
        }
        for(auto& it : s) it+=5;
    }
    for(int i=0;i<found.size() && i<4;i++){
        cout<<"? "<<string(5,found[i])<<endl;
        string resp;
        cin>>resp;
        for(int j=0;j<5;j++){
            if(resp[j]!='W'){
                ans[j]=found[i];
            }
        }
    }
    if(found.size()==5) for(auto& it : ans) if(it=='z') it=found[4];

    cout<<"! "<<ans<<endl;
}
int main()
{
    //assert(0);
    int t; cin>>t; while(t--)
        tc();
    return 0;
}

File: tasks/2025/round1/wordle2.py
#!/usr/bin/env python3
def tc():
    s = "abcde"
    ans = ["z"] * 5
    found = []

    # First loop to identify letters based on responses
    for iter in range(5):
        print(f"? {s}")
        resp = input().strip()  # Read response from input

        for i in range(5):
            if resp[i] != 'W':
                found.append(chr(ord('a') + iter * 5 + i))
        
        # Increment each character in `s` by 5
        s = ''.join(chr(ord(c) + 5) for c in s)

    # Second loop to refine answer based on found characters
    for i in range(min(len(found), 4)):
        print(f"? {found[i] * 5}")
        resp = input().strip()

        for j in range(5):
            if resp[j] != 'W':
                ans[j] = found[i]

    # If found has 5 elements, use the 5th element for any 'z' in ans
    if len(found) == 5:
        ans = [found[4] if ch == 'z' else ch for ch in ans]

    print(f"! {''.join(ans)}")

def main():
    t = int(input().strip())  # Number of test cases
    for _ in range(t):
        tc()

if __name__ == "__main__":
    main()
